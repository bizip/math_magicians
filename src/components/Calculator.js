import React from 'react';
import classes from './Calculator.module.css';
// import calculate from './logic/calculate';
// import operate from './logic/operate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  handleClick = (e) => {
    console.log(e.target.value);
  };

  render() {
    // const { name } = this.state;
    // console.log(name);
    return (
      <div className={classes.calc__grid}>
        <div className={`${classes.calc__btn} ${classes.btn__1} ${classes.gray}`}><h1 id="output" className={classes.white}>0</h1></div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            value="AC"
            onClick={this.handleClick}
          >
            AC
          </button>
        </div>

        <div className={classes.calc__btn}>
          <button type="button" onClick={this.handleClick} value="+/-">+/-</button>
        </div>

        <button type="button" onClick={this.handleClick} value="%">%</button>
        <div className={`${classes.calc__btn} ${classes.chockolate}`}>
          <button
            type="button"
            onClick={this.handleClick}
            value="%"
          >
            %
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="7"
          >
            7
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="8"
          >
            8
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="9"
          >
            9
          </button>
        </div>
        <div className={`${classes.calc__btn} ${classes.chockolate}`}>
          <button
            type="button"
            onClick={this.handleClick}
            value="*"
          >
            *
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="4"
          >
            4
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="5"
          >
            5
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="6"
          >
            6
          </button>
        </div>

        <div className={`${classes.calc__btn} ${classes.chockolate}`}><h1>-</h1></div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="1"
          >
            1
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="2"
          >
            2
          </button>
        </div>
        <div className={classes.calc__btn}>
          <button
            type="button"
            onClick={this.handleClick}
            value="3"
          >
            3
          </button>
        </div>
        <div className={`${classes.calc__btn} ${classes.chockolate}`}><h1>+</h1></div>
        <div className={`${classes.calc__btn} ${classes.zero}`}>
          <button
            type="button"
            onClick={this.handleClick}
            value="0"
          >
            0
          </button>
        </div>

        <div className={classes.calc__btn}><button type="button">.</button></div>

        <div className={`${classes.calc__btn} ${classes.chockolate}`}><button type="button">=</button></div>

      </div>
    );
  }
}

export default Calculator;
