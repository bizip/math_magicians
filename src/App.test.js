import renderer from 'react-test-renderer';
import Calculator from './components/Calculator';

describe('Test for all component', () => {
  it('It should render calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
