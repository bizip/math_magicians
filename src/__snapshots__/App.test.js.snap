// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for all component It should render calculator component 1`] = `
<div
  className="calc__grid"
>
  <div
    className="calc__btn btn__1 gray"
  >
    <h1
      className="white"
      id="output"
    />
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
  </div>
  <button
    onClick={[Function]}
    type="button"
    value="%"
  >
    %
  </button>
  <div
    className="calc__btn chockolate"
  >
    <button
      onClick={[Function]}
      type="button"
      value="/"
    >
      /
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
  </div>
  <div
    className="calc__btn chockolate"
  >
    <button
      onClick={[Function]}
      type="button"
      value="x"
    >
      *
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
  </div>
  <div
    className="calc__btn chockolate"
  >
    <button
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
  </div>
  <div
    className="calc__btn chockolate"
  >
    <button
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
  </div>
  <div
    className="calc__btn zero"
  >
    <button
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
  </div>
  <div
    className="calc__btn"
  >
    <button
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
  </div>
  <div
    className="calc__btn chockolate"
  >
    <button
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
